File: backend/ecommerce-service/src/main/java/com/travelplan/ecommerce/entity/Product.java
package com.travelplan.ecommerce.entity;

import com.travelplan.ecommerce.enums.ProductCategory;
import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "products")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;        
    
    @Column(columnDefinition = "TEXT")
    private String description; 
    
    @Enumerated(EnumType.STRING)
    @Column(length = 100)
    private ProductCategory category;    
    
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;
    
    @Column(name = "stock_quantity")
    private Integer stockQuantity = 0;
    
    @Column(name = "is_physical")
    private Boolean isPhysical = true;
    
    @Column(name = "event_id")
    private String eventId;

    // Handles multiple images in a separate table automatically
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "product_images", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "image_url")
    private List<String> images;    

    @Column(name = "is_active")
    private Boolean isActive = true;

    // Retained from previous fixes so your Mapper doesn't crash
    @Column(name = "average_rating", precision = 3, scale = 2)
    private BigDecimal averageRating = BigDecimal.ZERO;

    @Column(name = "review_count")
    private Integer reviewCount = 0;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/product.ts
export type ProductCategory = 'SOUVENIRS' | 'CLOTHING' | 'FOOD' | 'CRAFTS' | 'JEWELRY' | 'ART' | 'BOOKS' | 'OTHER'

export interface ProductDTO {
  id: number
  name: string
  category: string
  description?: string
  price: number
  stockQuantity: number
  imageUrl?: string
  averageRating: number
  reviewCount: number
  isActive: boolean
}

export interface CreateProductRequest {
  name: string
  category: ProductCategory
  description?: string
  price: number
  stockQuantity: number
}

export interface ProductSearchParams {
  category?: string
  minPrice?: number
  maxPrice?: number
  eventId?: string
}

export const PRODUCT_CATEGORIES: ProductCategory[] = [
  'SOUVENIRS',
  'CLOTHING',
  'FOOD',
  'CRAFTS',
  'JEWELRY',
  'ART',
  'BOOKS',
  'OTHER',
]
